### ISSUE TITLE
Dias Set uneven distribution

### ISSUE THREAD

**Issue by dirkmc** [ipfs-shipyard/peer-star-app#99]
I created a simulator to study the emergent behaviour of the Dias set. It appears there may be a bug in how the connections are distributed. Here's a video of the simulator:
https://www.youtube.com/watch?v=JB0rmXHLFV8
Note that the node at the top keeps receiving incoming connections from every other node, as the ring grows.

This PR adds a test case demonstrating the problem: https://github.com/ipfs-shipyard/peer-star-app/pull/98
To run the test:
`npx aegir test -t node -f test/common/dias-set-inbound-conns.js`

Example output:
```
...
Of 20 peers:
Max inbound: 19
Max outbound: 6
    1) distributes inbound connections evenly


  0 passing (3s)
  1 failing

  1) dias set inbound connections
       distributes inbound connections evenly:

      AssertionError: expected 19 to be below 15
      + expected - actual

      -19
      +15
```

**Comment by dirkmc** [comment]
I modified the simulator so that it can display nodes as they would be distributed around the ring organically (instead of evenly spaced):
https://youtu.be/GyFENcNGFNk

@pgte if my simulation is rendering the nodes correctly, it looks like maybe the calculation for the peer in position `node after 1/2 the ring` is incorrect?

This is the calculation I'm doing for where that node should be on the ring. Is `bytes.readUInt32BE()` correct for a 32 byte beer id (after removing the 2 byte prefix)?
```
  getPos(peer) {
    const bytes = peer.peerInfo.id.toBytes().slice(options.preambleByteCount)
    const proportion = bytes.readUInt32BE() / Math.pow(2, options.peerIdByteCount)
    const angle = 2 * Math.PI * proportion - Math.PI / 2
    const center = this.getCenter()
    const radius = this.getRadius() - this.pointRadius
    const x = center[0] + radius * Math.cos(angle)
    const y = center[1] + radius * Math.sin(angle)
    return [x, y]
  }
```

**Comment by dirkmc** [comment]
Also happy to put my simulator into a PR so you can play with it (it's pretty hacked together at the moment)

**Comment by dirkmc** [comment]
Ah I think it's a bug in the `oneHalfFrom` function:
```
  function oneHalfFrom (id) {
    const half = Buffer.alloc(bytes)
    for (let i = 1; i < bytes; i++) {
      half[i] = 0xff
    }
    half[0] = 0x7f
    return sum(bytes, half)
  }
```
Looks like it's not taking the `id` parameter into account

**Comment by pgte** [comment]
(I wasn't very trustful of all the byte manipulation that was going on, so I shoved a big-integer into that.
Happy to replace it with some clever byte-fu if you have it :) ).

**Comment by dirkmc** [comment]
@pgte oh great you fixed it already :)

**Comment by dirkmc** [comment]
I agree that it's better to use big-integer, it's possible that it's marginally less efficient, but I think it's much clearer and less likely to have bugs 👍

**Comment by pgte** [comment]
Fix landed in v0.9.2
